processor/deltatocumulativeprocessor/internal/lineartelemetry/metrics.go

```diff
@@ -39,19 +38,19 @@ type Metrics struct {
 	tracked func() int
 }
 
-func (m Metrics) Datapoints() Counter {
-	return Counter{Int64Counter: m.DeltatocumulativeDatapointsLinear}
+func (m Metrics) Datapoints() metric.Int64Counter {
+	return m.DeltatocumulativeDatapointsLinear
 }
 
 func (m *Metrics) WithTracked(streams func() int) {
 	m.tracked = streams
 }
 
-func Error(msg string) attribute.KeyValue {
-	return attribute.String("error", msg)
+func Error(msg string) metric.MeasurementOption {
```

Comment:
why does returning a `MeasurementOption` instead of a `KeyValue` and constructing the option later save allocations?